commandlineparser/src/Parser.cpp

```diff
@@ -130,3 +137,85 @@ void cmdParser::Parser::long_help(const std::vector<std::string>&keys) const
 
 	print(keys, print_long_help,"Long_Description");
 }
+
+void cmdParser::Parser::extract_value_as_string(int argc, char**argv)
+{ 
+	std::string key;
+	for (int i = 1; i < argc; i++)
+	{
+		if (!(strcmp(argv[i], "--help") == 0 || strcmp(argv[i], "-h") == 0)&&(!(strcmp(argv[i-1], "--help") == 0 || strcmp(argv[i-1], "-h") == 0)))
+		{
+			if (argv[i][0] == '-' || argv[i][1] == '-')
+			{
+				std::string temp(argv[i]);
+				size_t found = temp.find("=");
+				if (found != std::string::npos)
+				{
+					key = temp.substr(0, found);
+					std::string val = temp.substr(temp.find("=") + 1);
+
+					//validity checking block.
+					if (command_list.find(key) == command_list.end())
```

Comment:
optimize this.. same condition check two times

Reply:
done.